### ISSUE TITLE
Fetch election data for a given year from MongoDB

### ISSUE THREAD

**Issue by kymckay** [kymckay/f28cd-project#55]
The client-server communication structure is already done in #52 and this will build upon the code there.

I've set up the code structure for database querying in #54 

Essentially:

- The `database.js` file needs methods to get all of the data needed for a specific year
- The `server.js` file will use the methods in both the `/year` and `/years` request handlers to fetch the data for a specific year and then populate the response payload with it (in the same way I put the array of years into the payload of `/years`).
- I've set out some of the payload structure, but we can always tweak it as needed for convenience on the front end
- You'll need to inspect the scripts under the `data/` directory to see the document structure in the database (e.g. think the `candidate` collection will need to be joined with the `people` collection when querying to get their names).
- You'll need to reference the MongoDB JavaScript driver documentation to see the available methods for querying (they're very similar to querying from the command line).
- You may need to read up a bit on the `async` and `await` JavaScript keywords, but I think from the existing code you can tell how they're working on a surface level which should be enough for this (understanding the underlying Promise behaviour can be a bit tricky)

Code references:
https://github.com/kymckay/f28cd-project/blob/23f3bcc0fcfbd02c13514ab55d1f68e3180563a4/server/server.js#L37-L52

Here you see how the client request is processed on the server

https://github.com/kymckay/f28cd-project/blob/23f3bcc0fcfbd02c13514ab55d1f68e3180563a4/server/database.js#L20-L22

Here you see the actual query being executed. It's a relatively trivial one compared to some of the others that will be needed, but none of them should be too complex (I think the joining mentioned above will be the most).

**Comment by kymckay** [comment]
@jonleesy You mentioned some interest in this area of development

I've written the issue out in a way that should cover what you'll need to know, but I'm also happy to do this one.

I'd recommend just breaking development down into separate pull requests for each of the payload structures (e.g. just get the parties data first) and it should be more approachable that way.

Probably wait until the two PRs I have are merged in before starting.
